src/BackgroundProcess/Application/Port/Out/BackgroundJobQueue.php

```diff
@@ -0,0 +1,29 @@
+<?php
+declare(strict_types=1);
+
+namespace Jetty\BackgroundProcessing\BackgroundProcess\Application\Port\Out;
+
+/**
+ * Describes the minimum functionality needed for an async request.
+ */
+interface BackgroundJobQueue extends AsyncRequest
+{
+    /**
+     * Push an array of job data to the queue.
+     *
+     * @param array $data Data.
+     */
+    public function pushToQueue(array $data): BackgroundJobQueue;
+
+
+    /**
+     * Save the queue to be processed later.
+     */
+    public function save(): BackgroundJobQueue;
+
+
+    /**
+     * Cancel the background process.
+     */
+    public function cancel(): void;
+}
```

Comment:
I'm thinking each one of these methods should be documented as throwing an AsyncException. Right now, if the repository fails, we are swallowing that exception, and throwing it into a log somewhere. However, the caller may want to know about this. Especially with something like Bulk Emails: we really should know if the background process couldn't be started.

Reply:
I was thinking the same thing for some of them. I guess I'm not so sure that `AsyncException` would apply to this specific function. Maybe that's just because the name doesn't really fit to me.